#Seleccion de dataset para proyecto final_ autor: Luciano Benjamin Taddeo Córdoba - Comision 46270 - Tutor Rodrigo Alegre

Importaciones a utilizar

In [1]:
#Dataset 1|2|3
#Importar Pandas para procesar los datos de un archivo CSV
import pandas as pd
import numpy as np

#Dataset 1: Visualizacion Global de Migrantes Perdidos

In [2]:
#Carga el archivo con Pandas
df = pd.read_csv("Global Missing Migrants Dataset.csv")

In [3]:
#Obtener la propiedades del dataframe (filas, columnas)
df.shape

(13020, 19)

In [4]:
#Obtener información sobre el DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13020 entries, 0 to 13019
Data columns (total 19 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Incident Type                        13020 non-null  object 
 1   Incident year                        13020 non-null  int64  
 2   Reported Month                       13020 non-null  object 
 3   Region of Origin                     12998 non-null  object 
 4   Region of Incident                   13020 non-null  object 
 5   Country of Origin                    13012 non-null  object 
 6   Number of Dead                       12470 non-null  float64
 7   Minimum Estimated Number of Missing  13020 non-null  int64  
 8   Total Number of Dead and Missing     13020 non-null  int64  
 9   Number of Survivors                  13020 non-null  int64  
 10  Number of Females                    13020 non-null  int64  
 11  Number of Males             

In [5]:
#Tipos de datos del dataset
tiposdedatos = df.dtypes
print(tiposdedatos)

Incident Type                           object
Incident year                            int64
Reported Month                          object
Region of Origin                        object
Region of Incident                      object
Country of Origin                       object
Number of Dead                         float64
Minimum Estimated Number of Missing      int64
Total Number of Dead and Missing         int64
Number of Survivors                      int64
Number of Females                        int64
Number of Males                          int64
Number of Children                       int64
Cause of Death                          object
Migration route                         object
Location of death                       object
Information Source                      object
Coordinates                             object
UNSD Geographical Grouping              object
dtype: object


In [6]:
# Contar el número de valores únicos en cada columna de un DataFrame
unique_values = df.nunique()
unique_values

Incident Type                             4
Incident year                            10
Reported Month                           12
Region of Origin                         33
Region of Incident                       16
Country of Origin                       335
Number of Dead                           86
Minimum Estimated Number of Missing     120
Total Number of Dead and Missing        137
Number of Survivors                     228
Number of Females                        48
Number of Males                          59
Number of Children                       20
Cause of Death                           15
Migration route                          25
Location of death                      7460
Information Source                     3803
Coordinates                            9582
UNSD Geographical Grouping               19
dtype: int64

In [7]:
# Acceder a la primera fila utilizando iloc
primera_fila = df.iloc[0]

# Imprimir la primera fila
print(primera_fila)

Incident Type                                                                   Incident
Incident year                                                                       2014
Reported Month                                                                   January
Region of Origin                                                         Central America
Region of Incident                                                         North America
Country of Origin                                                              Guatemala
Number of Dead                                                                       1.0
Minimum Estimated Number of Missing                                                    0
Total Number of Dead and Missing                                                       1
Number of Survivors                                                                    0
Number of Females                                                                      0
Number of Males      

In [8]:
# Calcular la cantidad de registros por cada tipo de 'Flight Status'
TipodeIncidentes= df['Incident Type'].value_counts()

print(TipodeIncidentes)

Incident Type
Incident                   12670
Split Incident               261
Cumulative Incident           84
Incident,Split Incident        5
Name: count, dtype: int64


In [9]:
# Calcular el promedio de la columna 'Numero de Muertes'
PromMuertes = df['Number of Dead'].mean()

print("Promedio de Muertes:", PromMuertes)

Promedio de Muertes: 2.62373696872494


In [10]:
# Buscar valores iguales a 0 en todo el Dataframe
valores_igual_a_0 = df == 0
# Verificar si hay al menos un valor igual a 0 en todo el DataFrame
hay_0 = valores_igual_a_0.any().any()

# Contar cuántos valores iguales a 0 hay en todo el DataFrame
cantidad_de_0 = valores_igual_a_0.sum().sum()


print("¿Hay al menos un valor igual a 0 en todo el DataFrame?", hay_0)
print("Cantidad de valores iguales a 0 en todo el DataFrame:", cantidad_de_0)


¿Hay al menos un valor igual a 0 en todo el DataFrame? True
Cantidad de valores iguales a 0 en todo el DataFrame: 49877


De acuerdo con este Dataset, podremos realizar los siguientes analisis: 
Resumen estadístico de las columnas numéricas (por ejemplo, estadísticas descriptivas de las columnas de número de muertos, desaparecidos, sobrevivientes, etc.).

* Análisis de tendencias a lo largo de los años en relación con el tipo de incidente, la causa de la muerte, etc.

* Distribución geográfica de los incidentes utilizando las coordenadas geográficas.

* Identificación de las fuentes de información más frecuentes.

* Análisis de género y edad en relación con los incidentes.

* Comprobación de la relación entre la región de origen y la región del incidente.

#Dataset 2: Detección de Fraude en Tarjetas de Crédito

In [11]:
#Carga el archivo con Pandas
df2 = pd.read_csv("creditcard.csv")

In [12]:
#Obtener la propiedades del dataframe (filas, columnas)
df2.shape

(284807, 31)

In [13]:
#Obtener información sobre el DataFrame
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284807 entries, 0 to 284806
Data columns (total 31 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   Time    284807 non-null  float64
 1   V1      284807 non-null  float64
 2   V2      284807 non-null  float64
 3   V3      284807 non-null  float64
 4   V4      284807 non-null  float64
 5   V5      284807 non-null  float64
 6   V6      284807 non-null  float64
 7   V7      284807 non-null  float64
 8   V8      284807 non-null  float64
 9   V9      284807 non-null  float64
 10  V10     284807 non-null  float64
 11  V11     284807 non-null  float64
 12  V12     284807 non-null  float64
 13  V13     284807 non-null  float64
 14  V14     284807 non-null  float64
 15  V15     284807 non-null  float64
 16  V16     284807 non-null  float64
 17  V17     284807 non-null  float64
 18  V18     284807 non-null  float64
 19  V19     284807 non-null  float64
 20  V20     284807 non-null  float64
 21  V21     28

In [14]:
# Verifica el nombre de la columna y asegúrate de que sea correcto
print(df2.columns)

Index(['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount',
       'Class'],
      dtype='object')


In [15]:
#Tipos de datos del dataset
tiposdedatos = df2.dtypes
print (tiposdedatos)

Time      float64
V1        float64
V2        float64
V3        float64
V4        float64
V5        float64
V6        float64
V7        float64
V8        float64
V9        float64
V10       float64
V11       float64
V12       float64
V13       float64
V14       float64
V15       float64
V16       float64
V17       float64
V18       float64
V19       float64
V20       float64
V21       float64
V22       float64
V23       float64
V24       float64
V25       float64
V26       float64
V27       float64
V28       float64
Amount    float64
Class       int64
dtype: object


In [16]:
# Acceder a la primera fila utilizando iloc
primera_fila = df2.iloc[0]

# Imprimir la primera fila
print(primera_fila)

Time        0.000000
V1         -1.359807
V2         -0.072781
V3          2.536347
V4          1.378155
V5         -0.338321
V6          0.462388
V7          0.239599
V8          0.098698
V9          0.363787
V10         0.090794
V11        -0.551600
V12        -0.617801
V13        -0.991390
V14        -0.311169
V15         1.468177
V16        -0.470401
V17         0.207971
V18         0.025791
V19         0.403993
V20         0.251412
V21        -0.018307
V22         0.277838
V23        -0.110474
V24         0.066928
V25         0.128539
V26        -0.189115
V27         0.133558
V28        -0.021053
Amount    149.620000
Class       0.000000
Name: 0, dtype: float64


In [17]:
# Contar el número de valores únicos en cada columna de un DataFrame
unique_values = df2.nunique()
unique_values

Time      124592
V1        275663
V2        275663
V3        275663
V4        275663
V5        275663
V6        275663
V7        275663
V8        275663
V9        275663
V10       275663
V11       275663
V12       275663
V13       275663
V14       275663
V15       275663
V16       275663
V17       275663
V18       275663
V19       275663
V20       275663
V21       275663
V22       275663
V23       275663
V24       275663
V25       275663
V26       275663
V27       275663
V28       275663
Amount     32767
Class          2
dtype: int64

In [18]:
# Calcular la cantidad de registros por cada tipo de 'Flight Status'
Time = df2['Time'].value_counts()

print(Time)

Time
163152.0    36
64947.0     26
68780.0     25
3767.0      21
3770.0      20
            ..
127750.0     1
62260.0      1
62259.0      1
127753.0     1
172792.0     1
Name: count, Length: 124592, dtype: int64


In [19]:
# Calcular el promedio de la columna 'Edad'
PromQ = df2['Amount'].mean()

print("Promedio Cantidad:", PromQ)

Promedio Cantidad: 88.34961925093133


In [20]:
# Buscar valores iguales a 0 en todo el Dataframe
valores_igual_a_0 = df2 == 0
# Verificar si hay al menos un valor igual a 0 en todo el DataFrame
hay_0 = valores_igual_a_0.any().any()

# Contar cuántos valores iguales a 0 hay en todo el DataFrame
cantidad_de_0 = valores_igual_a_0.sum().sum()


print("¿Hay al menos un valor igual a 0 en todo el DataFrame?", hay_0)
print("Cantidad de valores iguales a 0 en todo el DataFrame:", cantidad_de_0)

¿Hay al menos un valor igual a 0 en todo el DataFrame? True
Cantidad de valores iguales a 0 en todo el DataFrame: 286142


Con este dataset, lo que se puede analizar es conocer las transacciones fraudulentas con las tarjetas de creditopara para que que los clientes no se les cobre por articulos que no compraron.
Este conjunto de datos son de transacciones de tarjetas de credito de septiembre de 2013 presenta transacciones realizadas por los titulares que ocurrieron en dos días, donde tenemos 492 fraudes de 284,807 transacciones. 
El conjunto de datos está muy desequilibrado, la clase positiva (fraudes) representa el 0,172% de todas las transacciones.
Con las variables que contiene el dataset podremos realizar los siguientes análisis:
* Analisis de las transforaciones de PCA (una técnica de selección de características concreta que utiliza una transformación ortogonal para convertir un conjunto de observaciones de variables, posiblemente correlacionadas, en un conjunto más reducido de variables que ya no guardan correlación y que se conocen como componentes principales)
*  Características V1, V2, ... V28 son los componentes principales obtenidos con PCA, las únicas características que no se han transformado con PCA son 'Time' y 'Amount'. La función 'Tiempo' contiene los segundos transcurridos entre cada transacción y la primera transacción del conjunto de datos. La característica 'Cantidad' es la cantidad de transacción, esta característica se puede utilizar para el aprendizaje sensible a los costos dependiente de ejemplos. La característica 'Clase' es la variable de respuesta y toma el valor 1 en caso de fraude y 0 en caso contrario


#Dataset 3: Restrasos de Vuelos

In [21]:
#Carga el archivo con Pandas
df3 = pd.read_csv("Airline Dataset.csv")

In [22]:
#Obtener la propiedades del dataframe (filas, columnas)
df3.shape

(98619, 15)

In [23]:
#Obtener información sobre el DataFrame
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 98619 entries, 0 to 98618
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Passenger ID          98619 non-null  int64 
 1   First Name            98619 non-null  object
 2   Last Name             98619 non-null  object
 3   Gender                98619 non-null  object
 4   Age                   98619 non-null  int64 
 5   Nationality           98619 non-null  object
 6   Airport Name          98619 non-null  object
 7   Airport Country Code  98619 non-null  object
 8   Country Name          98619 non-null  object
 9   Airport Continent     98619 non-null  object
 10  Continents            98619 non-null  object
 11  Departure Date        98619 non-null  object
 12  Arrival Airport       98619 non-null  object
 13  Pilot Name            98619 non-null  object
 14  Flight Status         98619 non-null  object
dtypes: int64(2), object(13)
memory usage

In [24]:
#Tipos de datos del dataset
tiposdedatos = df3.dtypes
print (tiposdedatos)

Passenger ID             int64
First Name              object
Last Name               object
Gender                  object
Age                      int64
Nationality             object
Airport Name            object
Airport Country Code    object
Country Name            object
Airport Continent       object
Continents              object
Departure Date          object
Arrival Airport         object
Pilot Name              object
Flight Status           object
dtype: object


In [25]:
# Acceder a la primera fila utilizando iloc
primera_fila = df3.iloc[0]

# Imprimir la primera fila
print(primera_fila)

Passenger ID                       10856
First Name                        Edithe
Last Name                         Leggis
Gender                            Female
Age                                   62
Nationality                        Japan
Airport Name            Coldfoot Airport
Airport Country Code                  US
Country Name               United States
Airport Continent                    NAM
Continents                 North America
Departure Date                 6/28/2022
Arrival Airport                      CXF
Pilot Name                 Edithe Leggis
Flight Status                    On Time
Name: 0, dtype: object


In [26]:
# Contar el número de valores únicos en cada columna de un DataFrame
unique_values = df3.nunique()
unique_values

Passenger ID            59871
First Name               8437
Last Name               41658
Gender                      2
Age                        90
Nationality               240
Airport Name             9062
Airport Country Code      235
Country Name              235
Airport Continent           6
Continents                  6
Departure Date            364
Arrival Airport          9024
Pilot Name              98610
Flight Status               3
dtype: int64

In [27]:
# Calcular la cantidad de registros por cada tipo de 'Flight Status'
FlightStatus = df3['Flight Status'].value_counts()

print(FlightStatus)

Flight Status
Cancelled    32942
On Time      32846
Delayed      32831
Name: count, dtype: int64


In [28]:
# Calcular el promedio de la columna 'Edad'
PromAge = df3['Age'].mean()

print("Promedio Edad:", PromAge)

Promedio Edad: 45.50402052342855


In [29]:
# Buscar valores iguales a 0 en todo el DataFrame
valores_igual_a_0 = df3 == 0
# Verificar si hay al menos un valor igual a 0 en todo el DataFrame
hay_0 = valores_igual_a_0.any().any()

# Contar cuántos valores iguales a 0 hay en todo el DataFrame
cantidad_de_0 = valores_igual_a_0.sum().sum()


print("¿Hay al menos un valor igual a 0 en todo el DataFrame?", hay_0)
print("Cantidad de valores iguales a 0 en todo el DataFrame:", cantidad_de_0)

¿Hay al menos un valor igual a 0 en todo el DataFrame? False
Cantidad de valores iguales a 0 en todo el DataFrame: 0


Con estos datos, puedes realizar varios análisis, dependiendo de tus objetivos. Aquí tienes algunas ideas de análisis:

Perfil de pasajeros: Resumen estadístico de la edad y género de los pasajeros. Análisis de nacionalidades de los pasajeros.

Rutas de vuelo: Identificación de las rutas más comunes. Análisis de la relación entre el aeropuerto de salida y el aeropuerto de llegada.

Estadísticas de vuelo: Análisis del estado de los vuelos, como la cantidad de vuelos programados, en curso y completados.

Tendencias temporales:Análisis de las fechas de salida para identificar patrones estacionales o tendencias a lo largo del tiempo.

Información geográfica: Análisis de la distribución de los aeropuertos de salida por continente.

Pilotos:Estudio de los pilotos y la cantidad de vuelos realizados por cada uno.

Combinación de datos: Combinación de datos geográficos para analizar la nacionalidad de los pasajeros en relación con los países y continentes de salida.

Segmentación de pasajeros: Clasificación de pasajeros por grupos de edad para comprender mejor las preferencias de viaje de diferentes generaciones.

Análisis de la duración del vuelo: Cálculo de la duración promedio de los vuelos y análisis de la variabilidad.
